crates/bitwarden-json/src/command.rs

```diff
@@ -166,3 +181,40 @@ pub enum ProjectsCommand {
     ///
     Delete(ProjectsDeleteRequest),
 }
+
+#[cfg(feature = "mobile")]
+#[derive(Serialize, Deserialize, JsonSchema, Debug)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub enum MobileCommand {
+    Kdf(MobileKdfCommand),
+    Crypto(MobileCryptoCommand),
+}
+
+#[cfg(feature = "mobile")]
+#[derive(Serialize, Deserialize, JsonSchema, Debug)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub enum MobileKdfCommand {
+    /// Sets the KDF parameters that are received in the prelogin request
+    ///
+    SetKdfParams(KdfParamRequest),
+    /// > Requires having previously set the KDF parameters
+    /// Calculates the user master password hash based on the provided password
+    ///
+    /// Returns: String
+    ///
+    GetUserPasswordHash(PasswordHashRequest),
+}
+
+#[cfg(feature = "mobile")]
+#[derive(Serialize, Deserialize, JsonSchema, Debug)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub enum MobileCryptoCommand {
+    /// > Requires having previously set the KDF parameters
+    /// Decrypts the users keys and initializes the user crypto, allowing for the encryption/decryption of the users items
+    ///
+    InitUserCrypto(InitUserCryptoRequest),
+    /// > Requires having previously initialized the user crypto
+    /// Decrypts the users organization keys, allowing for the encryption/decryption of the items in the users organizations
+    ///
+    InitOrgCrypto(InitOrgCryptoRequest),
```

Comment:
Should this be a single method? I don't think there are any valid scenarios where we only have one of these keys.

Reply:
I made those to be one to one with how we handle things internally, but you are right that from a consumer pov the separation is not needed.